x-pack/plugins/apm/public/components/app/correlations/ml_latency_correlations.tsx

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import { sum } from 'd3-array';
+import {
+  EuiIcon,
+  EuiLink,
+  EuiBasicTableColumn,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiProgress,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useUrlParams } from '../../../context/url_params_context/use_url_params';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+import { FETCH_STATUS } from '../../../hooks/use_fetcher';
+import { CorrelationsChart } from './correlations_chart';
+import {
+  FILTER_ACTION_LABEL,
+  CorrelationsTable,
+  SelectedSignificantTerm,
+  EXCLUDE_ACTION_LABEL,
+  EXCLUDE_ACTION_DESCRIPTION,
+  FILTER_LABEL,
+  FILTER_DESCRIPTION,
+} from './correlations_table';
+import { useCorrelations } from './use_correlations';
+import { createHref, push } from '../../shared/Links/url_helpers';
+import { useUiTracker } from '../../../../../observability/public';
+import { roundToDecimalPlace } from '../../../../common/search_strategies/correlations/formatting_utils';
+
+const DEFAULT_PERCENTILE_THRESHOLD = 95;
+const isErrorMessage = (arg: unknown): arg is Error => {
+  return arg instanceof Error;
+};
+
+interface Props {
+  onClose: () => void;
+}
+
+interface MlCorrelationsTerms {
+  correlation: number;
+  ksTest: number;
+  fieldName: string;
+  fieldValue: string;
+}
+
+export function MlLatencyCorrelations({ onClose }: Props) {
```

Comment:
Should we keep the `ml` part of this function and file name, and in the interface name above.

Reply:
I think for 7.14 I'll keep this here for now for clarity. We can remove the legacy Latency tab as well as rename this file in the next release.